In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import datetime as dt
import seaborn as sns
import scipy.stats as stats

%matplotlib inline

sns.set(style="ticks")

### BUSINESS PROBLEM-1

In [2]:
loans_data = pd.read_csv("C:\\Users\\kavit\\Desktop\\Laxman- Analytix labs\\Python\\Python Case Study\\Case Study-7\\Original file\\LoansData.csv")

In [3]:
loans_data.head()

Amount.Requested  Amount.Funded.By.Investors Interest.Rate Loan.Length  \
0           20000.0                     20000.0         8.90%   36 months   
1           19200.0                     19200.0        12.12%   36 months   
2           35000.0                     35000.0        21.98%   60 months   
3           10000.0                      9975.0         9.99%   36 months   
4           12000.0                     12000.0        11.71%   36 months   

         Loan.Purpose Debt.To.Income.Ratio State Home.Ownership  \
0  debt_consolidation               14.90%    SC       MORTGAGE   
1  debt_consolidation               28.36%    TX       MORTGAGE   
2  debt_consolidation               23.81%    CA       MORTGAGE   
3  debt_consolidation               14.30%    KS       MORTGAGE   
4         credit_card               18.78%    NJ           RENT   

   Monthly.Income FICO.Range  Open.CREDIT.Lines  Revolving.CREDIT.Balance  \
0         6541.67    735-739               14.0                   14272.0   
1         4583.33    715-719               12.0                   11140.0   
2        11500.00    690-694               14.0                   21977.0   
3         3833.33    695-699               10.0                    9346.0   
4         3195.00    695-699               11.0                   14469.0   

   Inquiries.in.the.Last.6.Months Employment.Length  
0                             2.0          < 1 year  
1                             1.0           2 years  
2                             1.0           2 years  
3                             0.0           5 years  
4                             0.0           9 years

In [4]:
loans_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 14 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Amount.Requested                2499 non-null   float64
 1   Amount.Funded.By.Investors      2499 non-null   float64
 2   Interest.Rate                   2498 non-null   object 
 3   Loan.Length                     2500 non-null   object 
 4   Loan.Purpose                    2500 non-null   object 
 5   Debt.To.Income.Ratio            2499 non-null   object 
 6   State                           2500 non-null   object 
 7   Home.Ownership                  2499 non-null   object 
 8   Monthly.Income                  2499 non-null   float64
 9   FICO.Range                      2498 non-null   object 
 10  Open.CREDIT.Lines               2497 non-null   float64
 11  Revolving.CREDIT.Balance        2497 non-null   float64
 12  Inquiries.in.the.Last.6.Months  24

In [5]:
loans_data['Interest.Rate'] = loans_data['Interest.Rate'].str.replace('%', '').astype('float')

In [6]:
loans_data['Loan.Length'] = loans_data['Loan.Length'].str.replace('months', '').astype('int')

In [7]:
loans_data['Debt.To.Income.Ratio'] = loans_data['Debt.To.Income.Ratio'].str.replace('%', '').astype('float')

In [8]:
loans_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 14 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Amount.Requested                2499 non-null   float64
 1   Amount.Funded.By.Investors      2499 non-null   float64
 2   Interest.Rate                   2498 non-null   float64
 3   Loan.Length                     2500 non-null   int32  
 4   Loan.Purpose                    2500 non-null   object 
 5   Debt.To.Income.Ratio            2499 non-null   float64
 6   State                           2500 non-null   object 
 7   Home.Ownership                  2499 non-null   object 
 8   Monthly.Income                  2499 non-null   float64
 9   FICO.Range                      2498 non-null   object 
 10  Open.CREDIT.Lines               2497 non-null   float64
 11  Revolving.CREDIT.Balance        2497 non-null   float64
 12  Inquiries.in.the.Last.6.Months  24

In [9]:
loans_data.isna().sum()

Amount.Requested                   1
Amount.Funded.By.Investors         1
Interest.Rate                      2
Loan.Length                        0
Loan.Purpose                       0
Debt.To.Income.Ratio               1
State                              0
Home.Ownership                     1
Monthly.Income                     1
FICO.Range                         2
Open.CREDIT.Lines                  3
Revolving.CREDIT.Balance           3
Inquiries.in.the.Last.6.Months     3
Employment.Length                 77
dtype: int64

In [10]:
for i in loans_data.columns:
    if loans_data[i].dtype == 'O':
        loans_data[i].fillna(loans_data[i].mode()[0], inplace = True)
    else:
        loans_data[i].fillna(loans_data[i].mean(), inplace = True)

In [11]:
loans_data.isna().sum()

Amount.Requested                  0
Amount.Funded.By.Investors        0
Interest.Rate                     0
Loan.Length                       0
Loan.Purpose                      0
Debt.To.Income.Ratio              0
State                             0
Home.Ownership                    0
Monthly.Income                    0
FICO.Range                        0
Open.CREDIT.Lines                 0
Revolving.CREDIT.Balance          0
Inquiries.in.the.Last.6.Months    0
Employment.Length                 0
dtype: int64

##### a. Interest rate is varied for different loan amounts (Less interest charged for high loan amounts)


##### 1. Define Null Hypothesis, H0:

    Interest rates and loan amounts are not correlated

##### 2. Define Alternate Hypothesis, Ha:

    Interest rates and loan amounts are correlated

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    Spearsmen Rank

In [12]:
stats.spearmanr(loans_data['Interest.Rate'], loans_data['Amount.Funded.By.Investors'])

SpearmanrResult(correlation=0.2849201249348258, pvalue=6.630719995757239e-48)

##### 5. Rejection Rule:
    The new p value < 0.05
        Hence, we will reject H0!

##### 6. Business Conclusion:
    Correlation=0.28. Hence, there is a positive relationship between the Interest rates and loan amounts but the 
    relationship strength is weak.

##### b. Loan length is directly effecting interest rate

##### 1. Define Null Hypothesis, H0:

    Interest rates and loan length are not related

##### 2. Define Alternate Hypothesis, Ha:

    Interest rates and loan length are related

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    pearsonr

In [13]:
stats.pearsonr(loans_data['Loan.Length'], loans_data['Interest.Rate'])

(0.42417192416620575, 9.54959606435366e-110)

##### 5. Rejection Rule:
    The new p value < 0.05
        Hence, we will reject H0!

##### 6. Business Conclusion:
    There is a positive relationship between the Interest rates and loan length.

##### c. Interest rate varies for different purpose of loans

##### 1. Define Null Hypothesis, H0:

            u1 = u2 = u3 ...... = u(n)
                    or
    Interest rates for different purpose of loans are same

##### 2. Define Alternate Hypothesis, Ha:

            u1 != u2 != u3 ...... != u(n)
                    or
    Interest rates for different purpose of loans are different

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    F-Test or ANOVA

In [14]:
loans_data['Loan.Purpose'].value_counts()

debt_consolidation    1307
credit_card            444
other                  201
home_improvement       152
major_purchase         101
small_business          87
car                     50
wedding                 39
medical                 30
moving                  29
vacation                21
house                   20
educational             15
renewable_energy         4
Name: Loan.Purpose, dtype: int64

In [15]:
debt_consolidation_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'debt_consolidation', 'Interest.Rate']

credit_card_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'credit_card', 'Interest.Rate']

other_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'other', 'Interest.Rate']

home_improvement_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'home_improvement', 'Interest.Rate']

major_purchase_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'major_purchase', 'Interest.Rate']

small_business_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'small_business', 'Interest.Rate']

car_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'car', 'Interest.Rate']

wedding_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'wedding', 'Interest.Rate']

medical_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'medical', 'Interest.Rate']

moving_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'moving', 'Interest.Rate']

vacation_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'vacation', 'Interest.Rate']

house_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'house', 'Interest.Rate']

educational_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'educational', 'Interest.Rate']

renewable_energy_purpose = loans_data.loc[loans_data['Loan.Purpose'] == 'renewable_energy', 'Interest.Rate']

In [16]:
stats.f_oneway(debt_consolidation_purpose, credit_card_purpose, other_purpose, home_improvement_purpose, 
               major_purchase_purpose, small_business_purpose, car_purpose, wedding_purpose, medical_purpose, 
               moving_purpose, vacation_purpose, house_purpose, educational_purpose, renewable_energy_purpose)

F_onewayResult(statistic=7.4826387840974355, pvalue=1.1599665535989115e-14)

##### 5. Rejection Rule:
    The new p value < 0.05
        Hence, we will reject H0!

##### 6. Business Conclusion:
    Interest rates for different purpose of loans are different.

##### d. There is relationship between FICO scores and Home Ownership. It means that, People with owning home will have high FICO scores.

##### 1. Define Null Hypothesis, H0:

    FICO scores and Home Ownership are not related

##### 2. Define Alternate Hypothesis, Ha:

     FICO scores and Home Ownership are related

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    chi square test

In [17]:
fico_home_ownership_wise_obs = pd.crosstab(loans_data['FICO.Range'], loans_data['Home.Ownership'])

In [18]:
fico_home_ownership_wise_obs

Home.Ownership  MORTGAGE  NONE  OTHER  OWN  RENT
FICO.Range                                      
640-644                0     0      0    1     4
645-649                1     0      0    0     2
650-654                1     0      0    0     0
655-659                0     0      1    0     3
660-664               41     0      0   17    67
665-669               52     0      0   18    75
670-674               68     0      1   13    91
675-679               79     0      1   11    75
680-684               55     0      2    9    91
685-689               61     0      0   10    65
690-694               55     0      0   10    75
695-699               68     0      0   10    75
700-704               60     0      0   12    59
705-709               66     0      0   12    56
710-714               55     0      0   10    47
715-719               46     0      0    2    45
720-724               49     0      0    1    64
725-729               48     0      0    5    41
730-734               46     0      0    7    41
735-739               30     0      0    6    29
740-744               26     0      0    4    23
745-749               26     0      0    7    21
750-754               32     0      0    4    25
755-759               28     0      0    6    12
760-764               28     0      0    4    14
765-769               22     0      0    3    11
770-774               12     0      0    1     4
775-779               13     0      0    3     6
780-784               21     0      0    3     4
785-789               12     0      0    2     5
790-794               15     0      0    1     4
795-799                7     0      0    1     5
800-804                7     1      0    3     2
805-809                6     0      0    2     3
810-814                5     0      0    1     2
815-819                6     0      0    0     0
820-824                0     0      0    1     0
830-834                1     0      0    0     0

In [19]:
stats.chi2_contingency(fico_home_ownership_wise_obs)

(473.564802816852,
 1.0065160858472195e-35,
 148,
 array([[2.29600e+00, 2.00000e-03, 1.00000e-02, 4.00000e-01, 2.29200e+00],
        [1.37760e+00, 1.20000e-03, 6.00000e-03, 2.40000e-01, 1.37520e+00],
        [4.59200e-01, 4.00000e-04, 2.00000e-03, 8.00000e-02, 4.58400e-01],
        [1.83680e+00, 1.60000e-03, 8.00000e-03, 3.20000e-01, 1.83360e+00],
        [5.74000e+01, 5.00000e-02, 2.50000e-01, 1.00000e+01, 5.73000e+01],
        [6.65840e+01, 5.80000e-02, 2.90000e-01, 1.16000e+01, 6.64680e+01],
        [7.94416e+01, 6.92000e-02, 3.46000e-01, 1.38400e+01, 7.93032e+01],
        [7.62272e+01, 6.64000e-02, 3.32000e-01, 1.32800e+01, 7.60944e+01],
        [7.20944e+01, 6.28000e-02, 3.14000e-01, 1.25600e+01, 7.19688e+01],
        [6.24512e+01, 5.44000e-02, 2.72000e-01, 1.08800e+01, 6.23424e+01],
        [6.42880e+01, 5.60000e-02, 2.80000e-01, 1.12000e+01, 6.41760e+01],
        [7.02576e+01, 6.12000e-02, 3.06000e-01, 1.22400e+01, 7.01352e+01],
        [6.01552e+01, 5.24000e-02, 2.62000e-01, 1.

##### 5. Rejection Rule:
    The new p value < 0.05
        Hence, we will reject H0!

##### 6. Business Conclusion:
    FICO scores and Home Ownership are related.

### BUSINESS PROBLEM - 2

In [20]:
price_quotes = pd.read_csv("C:\\Users\\kavit\\Desktop\\Laxman- Analytix labs\\Python\\Python Case Study\\Case Study-7\\Original file\\Price_Quotes.csv")

In [21]:
price_quotes.head()

Order_Number  Barry_Price  Mary_Price
0             1          126         114
1             2          110         118
2             3          138         114
3             4          142         111
4             5          146         129

In [22]:
price_quotes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   Order_Number  12 non-null     int64
 1   Barry_Price   12 non-null     int64
 2   Mary_Price    12 non-null     int64
dtypes: int64(3)
memory usage: 416.0 bytes


##### We would like to assess if there is any difference in the average price quotes provided by Mary and Barry. 

##### 1. Define Null Hypothesis, H0:

             u1 = u2
                or
    Average price quotes provided by Mary and Barry are same

##### 2. Define Alternate Hypothesis, Ha:

             u1 != u2
                or
    Average price quotes provided by Mary and Barry are different

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    T test

In [23]:
stats.ttest_ind(price_quotes.Mary_Price, price_quotes.Barry_Price)

Ttest_indResult(statistic=-1.4147436739281787, pvalue=0.17114226132118285)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    There is no difference in the average price quotes provided by Mary and Barry.

### BUSINESS PROBLEM-3 

In [24]:
treatment_facility = pd.read_csv("C:\\Users\\kavit\\Desktop\\Laxman- Analytix labs\\Python\\Python Case Study\\Case Study-7\\Original file\\Treatment_Facility.csv")

In [25]:
treatment_facility.head()

Month Reengineer  Employee_Turnover       VAR4        VAR5
0      1      Prior             0.0000  24.390244   42.682927
1      2      Prior             6.0606  19.354839   25.806452
2      3      Prior            12.1212  35.087719  146.198830
3      4      Prior             3.3333  18.404908  110.429448
4      5      Prior            12.9032  17.964072   23.952096

In [26]:
treatment_facility.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Month              20 non-null     int64  
 1   Reengineer         20 non-null     object 
 2   Employee_Turnover  20 non-null     float64
 3   VAR4               20 non-null     float64
 4   VAR5               20 non-null     float64
dtypes: float64(3), int64(1), object(1)
memory usage: 928.0+ bytes


#####  Determine what effect, if any, the reengineering effort had on the incidence behavioral problems and staff turnover. i.e To determine if the reengineering effort changed the critical incidence rate. Isthere evidence that the critical incidence rate improved?

##### 1. Define Null Hypothesis, H0:

    Employee turnover is same post reengineering
    
##### 2. Define Alternate Hypothesis, Ha:

    Employee turnover is different post reengineering

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    T test

In [27]:
employee_turnover_prior_reengineering = treatment_facility.loc[treatment_facility.Reengineer == 'Prior', 'Employee_Turnover']

In [28]:
employee_turnover_prior_reengineering

0      0.0000
1      6.0606
2     12.1212
3      3.3333
4     12.9032
5      9.6774
6     11.7647
7     11.4286
8     23.0769
9     15.0000
10    10.0000
11    26.0870
12    11.1111
Name: Employee_Turnover, dtype: float64

In [29]:
employee_turnover_post_reengineering = treatment_facility.loc[treatment_facility.Reengineer == 'Post', 'Employee_Turnover']

In [30]:
employee_turnover_post_reengineering

13    26.6667
14    17.9487
15    12.8205
16    15.7895
17     0.0000
18    30.7692
19    26.8293
Name: Employee_Turnover, dtype: float64

In [31]:
stats.ttest_ind(employee_turnover_post_reengineering, employee_turnover_prior_reengineering)

Ttest_indResult(statistic=1.770306704753604, pvalue=0.09361109345535304)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    Employee turnover is same post reengineering.

##### 1. Define Null Hypothesis, H0:

    Critical incidence rate did not improve post reengineering
    
##### 2. Define Alternate Hypothesis, Ha:

    Critical incidence rate improved post reengineering

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    T test

In [32]:
CI_prior_reengineering = treatment_facility.loc[treatment_facility.Reengineer == 'Prior', 'VAR5']

In [33]:
CI_prior_reengineering

0      42.682927
1      25.806452
2     146.198830
3     110.429448
4      23.952096
5      47.058824
6       0.000000
7      25.000000
8     132.183908
9      16.574586
10     88.757396
11     31.746032
12     10.152284
Name: VAR5, dtype: float64

In [34]:
CI_post_reengineering = treatment_facility.loc[treatment_facility.Reengineer == 'Post', 'VAR5']

In [35]:
CI_post_reengineering

13    29.126214
14    21.008403
15    34.782609
16    19.900498
17     9.950249
18    25.531915
19    23.148148
Name: VAR5, dtype: float64

In [36]:
stats.ttest_ind(CI_post_reengineering, CI_prior_reengineering)

Ttest_indResult(statistic=-1.6279144253528646, pvalue=0.12091989189884149)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    Critical incidence rate did not improve post reengineering.

### BUSINESS PROBLEM-4

In [37]:
priority_assessment = pd.read_csv("C:\\Users\\kavit\\Desktop\\Laxman- Analytix labs\\Python\\Python Case Study\\Case Study-7\\Original file\\Priority_Assessment.csv")

In [38]:
priority_assessment.head()

Days Priority
0   3.3     High
1   7.9   Medium
2   0.3     High
3   0.7   Medium
4   8.6   Medium

In [39]:
priority_assessment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 642 entries, 0 to 641
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Days      642 non-null    float64
 1   Priority  642 non-null    object 
dtypes: float64(1), object(1)
memory usage: 10.2+ KB


##### We will focus on the prioritization system. If the system is working, then high priority jobs, on average, should be completed more quickly than medium priority jobs, and medium priority jobs should be completed more quickly than low priority jobs. Use the data provided to determine whether this is, in fact, occurring.

##### 1. Define Null Hypothesis, H0:

    Completion of high prority(in days) >= Completion of medium prority(in days) >= Completion of low prority(in days)

##### 2. Define Alternate Hypothesis, Ha:

    Completion of high prority(in days) < Completion of medium prority(in days) < Completion of low prority(in days)

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    F-Test or ANOVA

In [40]:
high_priority_days = priority_assessment.loc[priority_assessment.Priority == 'High', 'Days']

In [41]:
medium_priority_days = priority_assessment.loc[priority_assessment.Priority == 'Medium', 'Days']

In [42]:
low_priority_days = priority_assessment.loc[priority_assessment.Priority == 'Low', 'Days']

In [43]:
stats.f_oneway(high_priority_days, medium_priority_days, low_priority_days)

F_onewayResult(statistic=1.812311010076072, pvalue=0.16411459461716182)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    The prioritization system is not working.

### BUSINESS PROBLEM-5

In [44]:
films = pd.read_csv("C:\\Users\\kavit\\Desktop\\Laxman- Analytix labs\\Python\\Python Case Study\\Case Study-7\\Original file\\Films.csv")

In [45]:
films.head()

_rowstate_                    Movie  Gender Marital_Status  Sinage  \
0           0  Ferris Buellers Day Off  Female        Married     2.0   
1           0  Ferris Buellers Day Off  Female         Single     1.0   
2           0  Ferris Buellers Day Off    Male        Married     2.0   
3           0  Ferris Buellers Day Off  Female        Married     1.0   
4           0  Ferris Buellers Day Off  Female        Married     1.0   

   Parking  Clean  Overall  Age  Income Hear_About  
0      2.0    2.0      2.0  3.0     1.0          5  
1      1.0    1.0      1.0  2.0     1.0          5  
2      4.0    3.0      2.0  4.0     1.0          5  
3      3.0    2.0      2.0  4.0     1.0          5  
4      1.0    1.0      1.0  3.0     3.0          1

In [46]:
films.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 330 entries, 0 to 329
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   _rowstate_      330 non-null    int64  
 1   Movie           330 non-null    object 
 2   Gender          330 non-null    object 
 3   Marital_Status  328 non-null    object 
 4   Sinage          328 non-null    float64
 5   Parking         328 non-null    float64
 6   Clean           327 non-null    float64
 7   Overall         328 non-null    float64
 8   Age             328 non-null    float64
 9   Income          314 non-null    float64
 10  Hear_About      323 non-null    object 
dtypes: float64(6), int64(1), object(4)
memory usage: 28.5+ KB


##### What is the overall level of customer satisfaction?

In [47]:
films.Overall.mean()

1.6189024390243902

In [48]:
print('Overall level of customer satisfaction is', films.Overall.mean())

Overall level of customer satisfaction is 1.6189024390243902


##### What factors are linked to satisfaction?

In [49]:
films.Gender.replace(['1', '2'], ['Male', 'Female'], inplace = True)

films.Marital_Status.replace(['1', '2'], ['Married', 'Single'], inplace = True)

In [50]:
films.Marital_Status.replace('Slngle', 'Single', inplace = True)

In [51]:
for i in films.columns:
    if (i == 'Signage') | (i == 'Parking') | (i == 'Clean') | (i == 'Overall'):
        films[i].fillna(films[i].mean(), inplace = True)
    else:
        films[i].fillna(films[i].mode()[0], inplace = True)

In [52]:
films.isna().sum()

_rowstate_        0
Movie             0
Gender            0
Marital_Status    0
Sinage            0
Parking           0
Clean             0
Overall           0
Age               0
Income            0
Hear_About        0
dtype: int64

###### Relationship between Movie and Overall:

##### 1. Define Null Hypothesis, H0:

    Movie and Overall are not related.

##### 2. Define Alternate Hypothesis, Ha:

    Movie and Overall are related.

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    Spearsmen Rank


In [53]:
stats.spearmanr(films.Movie, films.Overall)

SpearmanrResult(correlation=-0.007351177042819739, pvalue=0.8941650361229679)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    Correlation=-0.007. Hence, there is no relationship between the Movie and Overall.

##### Relationship between Gender and Overall: 

##### 1. Define Null Hypothesis, H0:

    Gender and Overall are not related.

##### 2. Define Alternate Hypothesis, Ha:

    Gender and Overall are related.

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    Spearsmen Rank


In [54]:
stats.spearmanr(films.Gender, films.Overall)

SpearmanrResult(correlation=0.0710279612716169, pvalue=0.1980883459095658)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    Correlation=0.07. Hence, there is no relationship between the Gender and Overall.

##### Relationship between Marital Status and Overall:

##### 1. Define Null Hypothesis, H0:

    Marital Status and Overall are not related.

##### 2. Define Alternate Hypothesis, Ha:

    Marital Status and Overall are related.

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    Spearsmen Rank


In [55]:
stats.spearmanr(films.Marital_Status, films.Overall)

SpearmanrResult(correlation=0.020362208866462228, pvalue=0.7124761935373358)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    Correlation=0.02. Hence, there is no relationship between the Marital Status and Overall.

##### Relationship between Sinage and Overall:

##### 1. Define Null Hypothesis, H0:

    Sinage and Overall are not related.

##### 2. Define Alternate Hypothesis, Ha:

    Sinage and Overall are related.

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    pearsonr


In [56]:
stats.pearsonr(films.Sinage, films.Overall)

(0.381988889367887, 6.612536004743165e-13)

##### 5. Rejection Rule:
    The new p value < 0.05
        Hence, we will reject H0!

##### 6. Business Conclusion:
    There is a relationship between the Sinage and Overall.

##### Relationship between Parking and Overall:

##### 1. Define Null Hypothesis, H0:

    Parking and Overall are not related.

##### 2. Define Alternate Hypothesis, Ha:

    Parking and Overall are related.

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    pearsonr


In [57]:
stats.pearsonr(films.Parking, films.Overall)

(0.5156270849681953, 8.21140012558417e-24)

##### 5. Rejection Rule:
    The new p value < 0.05
        Hence, we will reject H0!

##### 6. Business Conclusion:
    There is a relationship between the Parking and Overall.

##### Relationship between Clean and Overall:

##### 1. Define Null Hypothesis, H0:

    Clean and Overall are not related.

##### 2. Define Alternate Hypothesis, Ha:

    Clean and Overall are related.

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    pearsonr


In [58]:
stats.pearsonr(films.Clean, films.Overall)

(0.34924829684835673, 6.709795278873013e-11)

##### 5. Rejection Rule:
    The new p value < 0.05
        Hence, we will reject H0!

##### 6. Business Conclusion:
    There is a relationship between the Clean and Overall.

##### Relationship between Age and Overall:

##### 1. Define Null Hypothesis, H0:

    Age and Overall are not related.

##### 2. Define Alternate Hypothesis, Ha:

    Age and Overall are related.

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    pearsonr


In [59]:
stats.pearsonr(films.Age, films.Overall)

(-0.031230554456022852, 0.5718585915103257)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    Correlation=-0.03. Hence, there is no relationship between the Age and Overall.

##### Relationship between Income and Overall:

##### 1. Define Null Hypothesis, H0:

    Income and Overall are not related.

##### 2. Define Alternate Hypothesis, Ha:

    Income and Overall are related.

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    pearsonr


In [60]:
stats.pearsonr(films.Income, films.Overall)

(-0.011289134583879918, 0.8381146476505247)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    Correlation=-0.011. Hence, there is no relationship between the Income and Overall.

##### Relationship between Hear About and Overall:

##### 1. Define Null Hypothesis, H0:

    Hear About and Overall are not related.

##### 2. Define Alternate Hypothesis, Ha:

    Hear About and Overall are related.

##### 3. Define CI & p value:

    CI = 95%
    p values = 0.05
    
##### 4. Perform the test:

    Spearsmen Rank


In [61]:
stats.spearmanr(films.Hear_About, films.Overall)

SpearmanrResult(correlation=0.05296647309358267, pvalue=0.33745486087388665)

##### 5. Rejection Rule:
    The new p value > 0.05
        Hence, Do not reject H0!

##### 6. Business Conclusion:
    Correlation=0.05. Hence, there is no relationship between the Hear About and Overall.

#### From the above analysis, we can conclude that the Overall satisfaction depends on Sinage, Parking & Clean.

##### What is the demographic profile of Film on the Rocks patrons?

In [62]:
films.Gender.value_counts()

Female    213
Male      117
Name: Gender, dtype: int64

In [63]:
films.Marital_Status.value_counts()

Single     230
Married    100
Name: Marital_Status, dtype: int64

In [64]:
films.Age.value_counts()

2.0    177
3.0    117
1.0     26
4.0     10
Name: Age, dtype: int64

In [65]:
films.Income.value_counts()

1.0    158
3.0     90
2.0     82
Name: Income, dtype: int64

#### The demographic profile of Film on the Rocks patrons are:

    1. Females(213) are more as compared to Males(117).
    2. Singles(230) are more as compared to Married(100).
    3. Most no. of people are from Age group 2.0(13-30).
    4. Most no. of people are from Income group 1.0(Less than $50,000).

##### In what media outlet(s) should the film series be advertised?

In [66]:
films.Hear_About.value_counts()

5      233
4       41
1       22
3       14
2       12
3,4      2
2,5      2
5,4      1
4,5      1
1,5      1
3,5      1
Name: Hear_About, dtype: int64

#### Most of the viewers are coming due to Word of Mouth(5). Word of Mouth is not a media, therefore the film should be advertised by Website(4).